test/java/seedu/addressbook/commands/EditCommandTest.java

```diff
@@ -0,0 +1,251 @@
+                Collections.emptySet());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+
+        emptyDisplayList = TestUtil.createList();
+
+        listWithEveryone = TestUtil.createList(johnDoe, janeDoe, davidGrant, samDoe);
+        listWithSurnameDoe = TestUtil.createList(johnDoe, janeDoe, samDoe);
+    }
+
+    @Test
+    public void editCommand_invalidName_throwsException() {
+        final String[] invalidNames = { "", " ", "[]\\[;]" };
+        for (String name : invalidNames) {
+            assertConstructingInvalidEditCmdThrowsException(name, Phone.EXAMPLE, true, Email.EXAMPLE, false,
+                    Address.EXAMPLE, true, EMPTY_STRING_SET, 1);
+        }
+    }
+
+    @Test
+    public void editCommand_invalidPhone_throwsException() {
+        final String[] invalidNumbers = { "", " ", "1234-5678", "[]\\[;]", "abc", "a123", "+651234" };
+        for (String number : invalidNumbers) {
+            assertConstructingInvalidEditCmdThrowsException(Name.EXAMPLE, number, false, Email.EXAMPLE, true,
+                    Address.EXAMPLE, false, EMPTY_STRING_SET, 2);
+        }
+    }
+
+    @Test
+    public void editCommand_invalidEmail_throwsException() {
+        final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@",
+                "@invalid@email", "invalid@email!", "!invalid@email" };
+        for (String email : invalidEmails) {
+            assertConstructingInvalidEditCmdThrowsException(Name.EXAMPLE, Phone.EXAMPLE, false, email, false,
+                    Address.EXAMPLE, false, EMPTY_STRING_SET, 3);
+        }
+    }
+
+    @Test
+    public void editCommand_invalidAddress_throwsException() {
+        final String[] invalidAddresses = { "", " " };
+        for (String address : invalidAddresses) {
+            assertConstructingInvalidEditCmdThrowsException(Name.EXAMPLE, Phone.EXAMPLE, true, Email.EXAMPLE,
+                    true, address, true, EMPTY_STRING_SET, 4);
+        }
+    }
+
+    @Test
+    public void editCommand_invalidTags_throwsException() {
+        final String[][] invalidTags = { { "" }, { " " }, { "'" }, { "[]\\[;]" }, { "validTag", "" },
+                { "", " " } };
+        for (String[] tags : invalidTags) {
+            Set<String> tagsToAdd = new HashSet<>(Arrays.asList(tags));
+            assertConstructingInvalidEditCmdThrowsException(Name.EXAMPLE, Phone.EXAMPLE, true, Email.EXAMPLE,
+                    true, Address.EXAMPLE, false, tagsToAdd, 1);
+        }
+    }
+
+    /**
+     * Asserts that attempting to construct an add command with the supplied
+     * invalid data throws an IllegalValueException
+     */
+    private void assertConstructingInvalidEditCmdThrowsException(String name, String phone,
+                                                                boolean isPhonePrivate, String email, boolean isEmailPrivate, String address,
```

Comment:
Try to keep the indentation adhere to our guideline, we use 8 spaces of indentation from previous line for line wrap.